File: src/main/java/com/sunshines/bookstore/Repository/RoleRepository.java
package com.sunshines.bookstore.Repository;

import com.sunshines.bookstore.Model.Role;
import com.sunshines.bookstore.Model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface RoleRepository extends JpaRepository<Role, Integer> {
    List<Role> findAll();
    Role findFirstByName(String name);
}

File: src/main/java/com/sunshines/classrooms/Repository/RoleRepository.kt
package com.sunshines.classrooms.Repository

import com.sunshines.classrooms.Model.Role
import org.springframework.data .jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface RoleRepository : JpaRepository<Role, Int> {
    override fun findAll(): List<Role>
    fun findFirstByName(name: String): Role
}
